api/route.go

```diff
@@ -81,6 +81,46 @@ func (server *Server) GetResourceData(resp http.ResponseWriter, req *http.Reques
 	server.JSONWrite(resp, http.StatusOK, response)
 }
 
+// GetResourceTrends return trends by resource type, id, region and metric
+func (server *Server) GetResourceTrends(resp http.ResponseWriter, req *http.Request) {
+	queryErrs := url.Values{}
+	params := mux.Vars(req)
+	resourceType := params["type"]
+	filters := map[string]string{}
+
+	resourceId := req.URL.Query().Get("resourceId")
+	if resourceId == "" {
+		queryErrs.Add("resourceId", "resourceId field is mandatory")
+	}
+
+	region := req.URL.Query().Get("region")
```

Comment:
Why region is mandatory? 
why not get dynamic filters from the UI and set in into the query? this will be more flexible. you can check the function `GetFilterQueryParamWithOutPrefix`

Reply:
the storage.GetResourceTrends is dynamic, this one is the server/route implementation.
The mapping is static (Data.ResourceID -> resourceId), so I thought it'll be better to do this per use case so the function won't break.

Reply:
fixed - no longer mandatory, now the query should go with filters like: ?filter_Data.ResourceID=foo&filter_Data.Region=bar